### PR TITLE
gomod: update zoekt to include ngram sort optimization

### PR DESCRIPTION
**Pull request by keegancsmith**
Main motivation is to see the effect on performance for attribution search.

Note that this included a bump in the otel version used since zoekt is using a new version. This had a bit of a cascading effect on our third party deps since they removed a package. So this bumps most 3rd party packages that directly interacted with otel.

The changed commits are
https://github.com/sourcegraph/zoekt/compare/7643f3b313a2...45f608ff95bd

- https://github.com/sourcegraph/zoekt/commit/a176bde1a3 go get -u -t ./...
- https://github.com/sourcegraph/zoekt/commit/e2e8aede00 Fix template documentation comments
- https://github.com/sourcegraph/zoekt/commit/25c1ea5177 all: observe missing Stats RegexpsConsidered and FlushReason
- https://github.com/sourcegraph/zoekt/commit/9abbb8b0d3 zoekt-indexserver: Prevent invalid config from causing an NPE
- https://github.com/sourcegraph/zoekt/commit/008a775ba8 zoekt-indexserver: use value format directive for bad conf warning
- https://github.com/sourcegraph/zoekt/commit/f9d3a0e2e4 zoekt: add fgprof for full profiling
- https://github.com/sourcegraph/zoekt/commit/3d0bdd5c9c remove ngram offset code
- https://github.com/sourcegraph/zoekt/commit/45f608ff95 sort ngrams before looking them up

Test Plan: tested in the zoekt repo. Our CI will handle the dep updates. I eyeballed them and they all look low risk.

### CODE REVIEW

**Review by stefanhengl — APPROVED** [review]

File: dev/sg/sg_start_test.go [hunk 1]
```diff
@@ -82,7 +82,8 @@ func TestStartCommandSet_InstallError(t *testing.T) {
 		"",
 		"💡 Installing 1 commands...",
 		"--------------------------------------------------------------------------------",
-		"Failed to build test-cmd-1: 'bash -c echo 'booting up horsegraph' && exit 1' failed: booting up horsegraph: exit status 1:",
+		"Failed to build test-cmd-1: 'bash -c echo 'booting up horsegraph' && exit 1' failed: booting up horsegraph",
+		": exit status 1:",
```
Comment (keegancsmith) [inline dev/sg/sg_start_test.go:86]: I do not know why I needed to update this test, I imagine some sort of tty dependency changed how it detects our test terminal and now newlines happen in a different place? Either way I assume this is a safe update, but cc @mrnugget who likely has context here.

**Comment by keegancsmith** [comment]
FYI I have been blocked since zoekt updated some dependencies... which meant sourcegraph updated some dependencies... which suddenly meant there is some bazel darkarts going on.

File: go.mod [hunk 2]
```diff
@@ -213,42 +213,42 @@ require (
 	github.com/xeonx/timeago v1.0.0-rc4
 	github.com/yuin/gopher-lua v0.0.0-20210529063254-f4c35e4016d9
 	go.etcd.io/bbolt v1.3.6
-	go.opentelemetry.io/collector v0.75.0
-	go.opentelemetry.io/collector/component v0.75.0
+	go.opentelemetry.io/collector v0.81.0 // indirect
```
Comment (bobheadxi) [inline go.mod:216]: Beat me to the upgrade 😁 

Note that this requires an upgrade in the service as well: https://github.com/sourcegraph/sourcegraph/pull/54969/files#diff-c4b85f9a40207f45cc97a051e50a08352bbe2302c44b65911deec373410c4ad1

Comment (keegancsmith) [reply go.mod:216]: thanks! Is there a test or something we can add to point out that these need to happen in sync?

**Pull request merged**